### PR TITLE
feat: Add customized container for Devfile 

### PR DESCRIPTION
**Pull request by mkuznyetsov**
### What does this PR do?
 Instead of using default UDI image with post-install steps, just use a custom image (.devfile.Dockerfile) with those steps incorporated in it. Image will be located at  `quay.io/eclipse/che-operator-tools:latest` (TODO final image name yet to be confirmed).
with every commit to `.devfile.Dockerfile` in main branch, a Github Actions CI workflow will trigger a rebuild of the said image.

### What issues does this PR fix or reference?

Devtools Week task

### How to test this PR?

```bash
cat > /tmp/patch.yaml <<EOF
<PATCH_CONTENT>
EOF

chectl server:deploy \
     --installer operator \
     --platform <PLATFORM_TO_DEPLOY> \
     --che-operator-image <CUSTOM_OPERATOR_IMAGE> \
     --che-operator-cr-patch-yaml /tmp/patch.yaml
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Development resource are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-development-resources)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by mkuznyetsov** [comment]
@l0rd `quay.io/eclipse/che-operator-tools` repository is a good name for final image?

File: devfile.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ metadata:
 components:
   - name: tooling-container
     container:
-      image: quay.io/devfile/universal-developer-image:ubi8-latest
+      image: quay.io/mkuznets/che-operator-tools:latest
```
Comment (l0rd) [inline devfile.yaml:20]: Personal quay repo should be avoided. Something like quay.io/eclipse/che-operator-dev would be better

File: .devfile.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+FROM quay.io/devfile/universal-developer-image:ubi8-latest
+
+#install Go 1.18
+RUN cd /tmp && wget https://go.dev/dl/go1.18.6.linux-amd64.tar.gz && \
+        mkdir $HOME/go1.18 && \
+        tar -xvzf go1.18.6.linux-amd64.tar.gz -C $HOME/go1.18 --strip-components 1 && \
+        if ! grep -q "export PATH=\$HOME/go1.18/bin:\$PATH" $HOME/.bashrc; then echo "export PATH=\$HOME/go1.18/bin:\$PATH" >> $HOME/.bashrc; fi
+# install CHECTL
```
Comment (l0rd) [inline .devfile.Dockerfile:8]:
```suggestion

# install chectl
```

**Review by l0rd — CHANGES_REQUESTED** [review]

File: devfile.yaml (hunk 1 above)
Comment (mkuznyetsov) [reply devfile.yaml:20]:
```suggestion
      image: quay.io/eclipse/che-operator-dev:latest
```

**Pull request merged**